### PR TITLE
Realm Operator

### PR DESCRIPTION
**Pull request by vmuzikar**
The aim of this PR is to remove any functionality related to the KC deployment and leave only bits responsible for managing Realm and related resources.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,219 +1,121 @@
...
+
+## Motivation
+
+To provide the best experience, the new Keycloak Operator will use a new approach to manage Keycloak resources, such as Realms, Clients and Users. This approach will leverage the [new storage architecture](https://www.keycloak.org/2022/07/storage-map.html) and future immutability options, making the CRs the declarative single source of truth. In comparison to the [legacy Operator](https://github.com/keycloak/keycloak-operator), this will bring high robustness, reliability and predictability to the whole solution.
+
+Before we would consider the new Keycloak Operator ready for leveraging CRs, we expect completing several features including but not
+limited to:
+
+* File store (expected in Keycloak 20) to persist data in a file instead of DB.
+* Read-only administration REST API, UI Console and other interfaces. This is required for the new immutability concept
```
Comment (stianst) [inline README.md:17]: I'd reword this to make it explicit that it's read-only for resources defined in the static store

File: README.md [hunk 2]
```diff
@@ -1,219 +1,121 @@
...
+as a side-car for it.
+
+## Motivation
+
+To provide the best experience, the new Keycloak Operator will use a new approach to manage Keycloak resources, such as Realms, Clients and Users. This approach will leverage the [new storage architecture](https://www.keycloak.org/2022/07/storage-map.html) and future immutability options, making the CRs the declarative single source of truth. In comparison to the [legacy Operator](https://github.com/keycloak/keycloak-operator), this will bring high robustness, reliability and predictability to the whole solution.
+
+Before we would consider the new Keycloak Operator ready for leveraging CRs, we expect completing several features including but not
+limited to:
+
+* File store (expected in Keycloak 20) to persist data in a file instead of DB.
```
Comment (stianst) [inline README.md:16]: I'd reword this to something like static/immutable store, and clarify that it should be supporting git flow, as well as CRs.

File: README.md [hunk 3]
```diff
@@ -1,219 +1,121 @@
...
+
+For details, please see the following chapters.
+
+## Using the Realm Import CR and  the new Keycloak Operator
+
+As the name suggests, the Realm Import CR is meant only for declarative Realm creation (no updates or deletion is supported), very much like the [Realm CR in the legacy Operator](https://github.com/keycloak/keycloak-operator/blob/main/deploy/crds/keycloak.org_keycloakrealms_crd.yaml) which was used the same way. The main advantage and difference of the Realm Import CR are that it contains the full Realm representation (incl. all sub-resources like Clients, Users, etc.) unlike the old Realm CR which included only selected fields.
+
+To learn more about the Realm Import CR, see the [documentation](https://www.keycloak.org/operator/realm-import).
+
+## Using the Realm Operator
```
Comment (stianst) [inline README.md:37]: I'm a bit confused here. Why mention the "Realm Operator" at all? I'd rather just make the new recommended thing the new Operator + realm manger operator, and leave the old operator completely out (as we won't do any updates to it after KC 20 is out).

File: README.md [hunk 4]
```diff
@@ -1,219 +1,121 @@
...
-# Legacy Keycloak Operator
-A Kubernetes Operator based on the Operator SDK for creating and syncing resources in Keycloak [WildFly distribution].
-
-**NOTE: This Operator is deprecated and will reach EOL with [Keycloak 20](https://www.keycloak.org/2022/03/releases).
-Consider migrating to the [new Keycloak Operator](https://github.com/keycloak/keycloak/tree/main/operator).**
+# Keycloak Realm Operator
+A Kubernetes Operator based on the Operator SDK for managing Realm and its sub-resources in Keycloak. This Operator is
+forked from the [legacy Keycloak Operator](https://github.com/keycloak/keycloak-operator) and stripped off of any functionality
+related to Keycloak Deployment as it was designed for the WildFly distribution of Keycloak and is not compatible with the
+new Quarkus distribution. The Realm Operator is meant as a **temporary workaround** to compensate for the missing
```
Comment (mabartos) [inline README.md:5]: @vmuzikar It'd be good to mention sth like that even in the description for these operators in Quay/DockerHub/OperatorHub, right? Probably could be done in a follow-up task, but IMHO it's quite important to have at least some description there. 

The thing is, I'm kinda afraid that could be a little bit confusing for the users as they see the `keycloak-operator` and `keycloak-realm-operator` in a hub together.

Comment (mabartos) [reply README.md:5]: Also, maybe add some guide to the keycloak-web??

Comment (vmuzikar) [reply README.md:5]: Sure, we'll have to prepare release notes for it. However, at this moment we don't plan to publish `keycloak-realm-operator` in OperatorHub as it is only a temporary workaround.

File: README.md [hunk 5]
```diff
@@ -1,219 +1,121 @@
...
+    ```
+    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-k8s-resources/19.0.1/kubernetes/keycloaks.k8s.keycloak.org-v1.yml
+    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-k8s-resources/19.0.1/kubernetes/keycloakrealmimports.k8s.keycloak.org-v1.yml
+    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-k8s-resources/19.0.1/kubernetes/kubernetes.yml
+    ```
+    Notice that we're using a hardcoded `19.0.1` version here. Feel free to use a newer version. See also the [installation guide](https://www.keycloak.org/operator/installation#_vanilla_kubernetes_installation).
+
+2.  Deploy the Realm Operator.
+    ```
+    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-realm-operator/main/deploy/crds/legacy.k8s.keycloak.org_externalkeycloaks_crd.yml
```
Comment (mabartos) [inline README.md:53]: Should be `.yaml` instead of `.yml` for the specified CRDs, right? Same below.

File: README.md [hunk 6]
```diff
@@ -1,219 +1,121 @@
...
+    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-realm-operator/main/deploy/crds/legacy.k8s.keycloak.org_keycloakclients_crd.yml
+    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-realm-operator/main/deploy/crds/legacy.k8s.keycloak.org_keycloakrealms_crd.yml
+    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-realm-operator/main/deploy/crds/legacy.k8s.keycloak.org_keycloakusers_crd.yml
+    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-realm-operator/main/deploy/role.yaml
+    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-realm-operator/main/deploy/role_binding.yaml
+    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-realm-operator/main/deploy/service_account.yaml
+    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-realm-operatpr/main/deploy/operator.yaml
```
Comment (mabartos) [inline README.md:60]:
```suggestion
    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-realm-operator/main/deploy/operator.yaml
```

File: README.md [hunk 7]
```diff
@@ -1,219 +1,121 @@
...
+    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-k8s-resources/19.0.1/kubernetes/kubernetes.yml
+    ```
+    Notice that we're using a hardcoded `19.0.1` version here. Feel free to use a newer version. See also the [installation guide](https://www.keycloak.org/operator/installation#_vanilla_kubernetes_installation).
+
+2.  Deploy the Realm Operator.
+    ```
+    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-realm-operator/main/deploy/crds/legacy.k8s.keycloak.org_externalkeycloaks_crd.yml
+    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-realm-operator/main/deploy/crds/legacy.k8s.keycloak.org_keycloakclients_crd.yml
+    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-realm-operator/main/deploy/crds/legacy.k8s.keycloak.org_keycloakrealms_crd.yml
+    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-realm-operator/main/deploy/crds/legacy.k8s.keycloak.org_keycloakusers_crd.yml
```
Comment (mabartos) [inline README.md:56]: Should be .yaml instead of .yml for the specified CRDs, right?

```suggestion
    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-realm-operator/main/deploy/crds/legacy.k8s.keycloak.org_externalkeycloaks_crd.yaml
    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-realm-operator/main/deploy/crds/legacy.k8s.keycloak.org_keycloakclients_crd.yaml
    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-realm-operator/main/deploy/crds/legacy.k8s.keycloak.org_keycloakrealms_crd.yaml
    kubectl apply -f https://raw.githubusercontent.com/keycloak/keycloak-realm-operator/main/deploy/crds/legacy.k8s.keycloak.org_keycloakusers_crd.yaml
```

File: README.md (hunk 6 above)
Comment (vmuzikar) [reply README.md]: Very good catch, thank you!

File: README.md (hunk 7 above)
Comment (vmuzikar) [reply README.md]: Very good catch, thank you!

File: README.md (hunk 2 above)
Comment (vmuzikar) [reply README.md:16]: I think we're stating the static/immutable in the next bullet point as the file store and static/immutable store are really two relatively separated things.

Comment (vmuzikar) [reply README.md:16]: @stianst Do you have anything specific how to reword this? :)

File: README.md (hunk 1 above)
Comment (vmuzikar) [reply README.md:17]: I added a mention of GitOps. Is it something like you had in mind?

**Comment by mabartos** [comment]
@vmuzikar Do you know why the GHAs weren't executed for PR? The new changes should be propagated and GHAs executed with them, right?

**Comment by mabartos** [comment]
@vmuzikar Very nice job! Thanks!

Code looks good to me, but it's necessary to verify the proper functionality of GHAs.

**Comment by mabartos** [comment]
Any progress here?

**Comment by vmuzikar** [comment]
@mabartos Found a button to actually enable GHA in this fork. :) CI is running now.

**Review by mabartos — APPROVED** [review]
@vmuzikar Ok, it seems everything works as expected. LGTM, thanks!

**Review by stianst — APPROVED** [review]
Readme looks good, and approving based on previous reviews
